ocaml/lambda/matching.ml

```diff
@@ -1188,8 +1205,9 @@ let what_is_cases = what_is_cases ~skip_any:true
 let pm_free_variables { cases } =
   List.fold_right
     (fun (_, act) r ->
-       Ident.Set.union (free_variables (lambda_of_action act)) r)
-    cases Ident.Set.empty
+       Option.bind r
+         (fun r -> Option.map (Ident.Set.union r) (free_variables_of_rhs act)))
+    cases (Some Ident.Set.empty)
```

Comment:
I think it would be clearer to inline `free_variables_of_rhs` here &mdash; its only use is this function, and the semantics of `None` vs. `Some` are clearer closer to the use site.